test/rubygems/test_gem_requirement.rb

```diff
@@ -35,7 +35,6 @@ def test_initialize
     assert_requirement_equal "= 2", ["2"]
     assert_requirement_equal "= 2", v(2)
     assert_requirement_equal "2.0", "2"
-    assert_requirement_equal ["= 2", ">= 2"], [">= 2", "= 2"]
```

Comment:
If this PR still fixes https://github.com/rubygems/rubygems/pull/2230 why we deleting this assertion?

Reply:
As I understand the wording of that PR, the problem was that versions `1.0.0` and `1.0` were not considered equivalent. Or similarly for [the associated bundler issue](https://github.com/bundler/bundler/issues/6295), that the requirements `>= 0.0.0` and `>= 0` were not considered equivalent.

#2230 fixed that issue and added a bunch of assertions covering that, but at the same time started sorting the requirements and added this specific test covering that.

In this PR I'm only reverting the sorting of requirements (and thus removing this associated spec), but I'm keep the fix for the above problem and all the tests covering that.

Reply:
@segiddins want to give feedback? ^

Reply:
I’m not happy to see this reverted

Reply:
Can you elaborate @segiddins? It's no big deal if we need to revert this PR.